tests/Integration/Behaviour/Features/Context/Domain/Discount/DiscountFeatureContext.php

```diff
@@ -177,6 +177,21 @@ public function createDiscount(string $discountReference, string $discountType,
             $command->setCustomerId($this->getSharedStorage()->get($data['customer']));
         }
 
+        if (isset($data['customer_groups'])) {
```

Comment:
Again this should be moved into conditions, but same global comment for the whole PR